lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -140,8 +140,23 @@ def add_stored_credentials(post, creditcard, options)
 
       def add_three_d_secure(post, options)
         params = post[:parametros] ||= {}
-
-        params[:ThreeDsResponse] = options[:three_d_secure] if options[:three_d_secure]
+        three_d_secure = options[:three_d_secure]
+        return unless three_d_secure.present?
+
+        three_d_response = {}
+        three_d_response[:exemption_type] = three_d_secure[:exemption_type]
+        three_d_response[:three_ds_version] = three_d_secure[:version]
+        three_d_response[:authentication_value] = three_d_secure[:cavv]
+        three_d_response[:directory_server_transaction_id] = three_d_secure[:ds_transaction_id]
+        three_d_response[:acs_transaction_id] = three_d_secure[:acs_transaction_id]
+        three_d_response[:authentication_response_status] = three_d_secure[:authentication_response_status]
+        three_d_response[:three_ds_server_trans_id] = three_d_secure[:three_ds_server_trans_id]
+        three_d_response[:ecommerce_indicator] = three_d_secure[:eci]
+
+        extra_params = options[:extra_options_for_three_d_secure].with_indifferent_access
```

Comment:
🔴  ❓ Question 🔴 

So the idea is that Cecabank is going to need a ton of values from the 3DS Response and you want to add a GSF to allow the merchant to provide the extra data, right?

If that is the idea, is not going to work, on 3DS Global the merchant doesn't have access to the actual 3DS response, the response is saved on CORE DB after the customer has finished the 3DS.

We need to understand what are the required fields on that `ThreeDsResponse`.

Reply:
I am going to test which fields are necessary, which fields we can obtain from the options, and which fields we can ask if they can be excluded in the Cecabank API

Reply:
I removed extra_options_for_three_d_secure because the only field that we need is amount and this field we have in the options